pkg/transformer/kubernetes/kubernetes_test.go

```diff
@@ -451,3 +453,39 @@ func TestUnsupportedKeys(t *testing.T) {
 	}
 
 }
+
+func TestRestartOnFailure(t *testing.T) {
+
+	kobjectWithRestartOnFailure := newKomposeObject()
+	serviceConfig := kobjectWithRestartOnFailure.ServiceConfigs["app"]
+	serviceConfig.Restart = "on-failure"
+	kobjectWithRestartOnFailure.ServiceConfigs = map[string]kobject.ServiceConfig{"app": serviceConfig}
+
+	// define all test cases for RestartOnFailure function
+	replicas := 2
+	testCase := map[string]struct {
+		komposeObject kobject.KomposeObject
+		opt           kobject.ConvertOptions
+	}{
+		// objects generated are deployment, service and replication controller
+		"Do not Create Deployment (D) with restart:'on-failure'":             {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsDeploymentFlag: true, Replicas: replicas}},
+		"Do not Create DaemonSet (DS) with restart:'on-failure'":             {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsDaemonSetFlag: true, Replicas: replicas}},
+		"Do not Create ReplicationController (RC) with restart:'on-failure'": {kobjectWithRestartOnFailure, kobject.ConvertOptions{IsReplicationControllerFlag: true, Replicas: replicas}},
+	}
+
+	for name, test := range testCase {
+		t.Log("Test case:", name)
+		k := Kubernetes{}
+		if os.Getenv("BE_CRASHER") == "1" {
```

Comment:
@kadel Can you check if this unit test works fine, so that I can remove [WIP] from it.